File: src/main/java/baboon/industry/recipe/fuel/AdvancedRedstoneFuel.java
package baboon.industry.recipe.fuel;

import net.minecraft.core.block.Block;
import net.minecraft.core.item.Item;

import java.util.HashMap;

public class AdvancedRedstoneFuel {
    private static final HashMap<Integer, Integer> redstoneList = new HashMap<>();

    public AdvancedRedstoneFuel() {
        addRedstoneFuel(Item.dustRedstone.id, 100);
        addRedstoneFuel(Block.blockRedstone.id, 900);
    }

    public static void addRedstoneFuel(int inputItem, int outputYield) {
        redstoneList.put(inputItem, outputYield);
    }

    public int getYield(int i) {
        return redstoneList.get(i);
    }

    public int getRedstoneFuel(int i) {
        return redstoneList.get(i) == null ? 0 : redstoneList.get(i);
    }

    public HashMap<Integer, Integer> getRedstoneList() {
        return redstoneList;
    }
}

File: src/main/kotlin/turniplabs/industry/recipes/fuels/AdvancedRedstoneFuel.kt
package turniplabs.industry.recipes.fuels

import net.minecraft.core.block.Block
import net.minecraft.core.item.Item

object AdvancedRedstoneFuel {
    private val redstoneList = HashMap<Any?, Any?>()

    init {
        addRedstoneFuel(Item.dustRedstone.id, 128)
        addRedstoneFuel(Block.blockRedstone.id, 1152)
    }

    fun addRedstoneFuel(input: Int, yield: Int) {
        redstoneList[input] = yield
    }

    fun getYield(i: Int): Int? {
        return redstoneList[i] as Int?
    }

    fun getRedstoneFuel(i: Int): Int? {
        return if (redstoneList[i] == null) 0 else redstoneList[i] as Int?
    }

    fun getRedstoneFuelList(): HashMap<Any?, Any?> {
        return redstoneList
    }
}